Packages/com.unity.inputsystem/InputSystem/Editor/InputEditorAnalytics.cs

```diff
@@ -0,0 +1,924 @@
+                sessionFocusDurationSeconds = 0;
+                sessionFocusDurationSeconds = 0;
+                sessionFocusSwitchCount = 0;
+                actionMapModificationCount = 0;
+                actionModificationCount = 0;
+                bindingModificationCount = 0;
+                explicitSaveCount = 0;
+                autoSaveCount = 0;
+                resetCount = 0;
+                controlSchemeModificationCount = 0;
+            }
+
+            /// <summary>
+            /// Specifies what kind of Input Actions editor this event represents.
+            /// </summary>
+            public Kind kind;
+
+            /// <summary>
+            /// The total duration for the session, i.e. the duration during which the editor window was open.
+            /// </summary>
+            public float sessionDurationSeconds;
+
+            /// <summary>
+            /// The total duration for which the editor window was open and had focus.
+            /// </summary>
+            public float sessionFocusDurationSeconds;
+
+            /// <summary>
+            /// Specifies the number of times the window has transitioned from not having focus to having focus in a single session.
+            /// </summary>
+            public int sessionFocusSwitchCount;
+
+            /// <summary>
+            /// The total number of action map modifications during the session.
+            /// </summary>
+            public int actionMapModificationCount;
+
+            /// <summary>
+            /// The total number of action modifications during the session.
+            /// </summary>
+            public int actionModificationCount;
+
+            /// <summary>
+            /// The total number of binding modifications during the session.
+            /// </summary>
+            public int bindingModificationCount;
+
+            /// <summary>
+            /// The total number of controls scheme modifications during the session.
+            /// </summary>
+            public int controlSchemeModificationCount;
+
+            /// <summary>
+            /// The total number of explicit saves during the session, i.e. as in user-initiated save.
+            /// </summary>
+            public int explicitSaveCount;
+
+            /// <summary>
+            /// The total number of automatic saves during the session, i.e. as in auto-save on close or focus-lost.
+            /// </summary>
+            public int autoSaveCount;
+
+            /// <summary>
+            /// The total number of user-initiated resets during the session, i.e. as in using Reset option in menu.
+            /// </summary>
+            public int resetCount;
+
+            public bool isValid => kind != Kind.Invalid && sessionDurationSeconds >= 0;
+
+            public override string ToString()
+            {
+                return $"{nameof(kind)}: {kind}, " +
+                    $"{nameof(sessionDurationSeconds)}: {sessionDurationSeconds} seconds, " +
```

Comment:
Since it wouldn't break if someone renamed the type in refactoring. The ToString() here is generated by Resharper which does it the right way, which I believe is this way. Seen many times where devs write strings and then someone refactors and then code is broken. Hence nameof is always preferable if reflecting actual field name. The ToString here is only for convenience while developing.